File: 2116_count_equal_and_divisible_pairs_in_an_array.java
class Solution {
    public int countPairs(int[] q, int p) {
        int c = 0;
        for (int u = 0; u < q.length - 1; u++)
            for (int v = u + 1; v < q.length; v++)
                c += (q[u] == q[v] && u * v % p == 0) ? 1 : 0;
        return c;
    }
}

File: 2116_count_equal_and_divisible_pairs_in_an_array.cpp
class Solution {
public:
    int countPairs(vector<int>& m, int q) {
        int s = 0;
        for(int i = 0; i < m.size(); ++i)
            for(int j = i + 1; j < m.size(); ++j)
                if(m[i] == m[j] && i * j % q < 1) ++s;
        return s;
    }
};
